### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by lindsayismith**
Add caching to the credential provider, make it a singleton, add periodic refreshes, and make gcloud handle expired auth more gracefully

### CODE REVIEW

**Review by Gridasov — APPROVED** [review]

File: artifactregistry-auth-common/src/main/java/com/google/cloud/artifactregistry/auth/DefaultCredentialProvider.java [hunk 1]
```diff
@@ -30,21 +32,55 @@ public class DefaultCredentialProvider implements CredentialProvider {
   private static String[] SCOPES = {"https://www.googleapis.com/auth/cloud-platform",
       "https://www.googleapis.com/auth/cloud-platform.read-only"};
 
-  @Override
+  private GoogleCredentials cachedCredentials;
+
+  // Singleton instance
+  private static DefaultCredentialProvider defaultInstance = new DefaultCredentialProvider();
```
[the remark this answers is not in the document]
Comment (lindsayismith) [reply artifactregistry-auth-common/src/main/java/com/google/cloud/artifactregistry/auth/DefaultCredentialProvider.java]: Done, thanks.

**Comment by lindsayismith** [comment]
@googlebot: rescan

**Pull request merged**